src/FormBuilder/resources/js/form-builder/src/components/EmbedForm/index.tsx

```diff
@@ -0,0 +1,542 @@
+
+        const attributes = {
+            id: formId
+        }
+
+        if (isButton) {
+            attributes['displayStyle'] = state.selectedStyle;
+            attributes['continueButtonTitle'] = state.openFormButton;
+        }
+
+        return `<!-- wp:give/donation-form ${JSON.stringify(attributes)} /-->`;
+    }
+
+    const getShortcode = () => {
+        const shortcodeAttributes = [];
+
+        const attributes = {
+            id: formId
+        }
+
+        if (isButton) {
+            attributes['display_style'] = state.selectedStyle;
+            attributes['continue_button_title'] = state.openFormButton;
+        }
+
+        for (const key in attributes) {
+            shortcodeAttributes.push(`${key}="${attributes[key]}"`);
+        }
+
+        return `[give_form ${shortcodeAttributes.join(' ')}]`;
+    }
+
+    /**
+     * Handle copying shortcode to clipboard
+     */
+    const handleCopy = async () => {
+        await navigator.clipboard.writeText(getShortcode());
+
+        setState(prevState => {
+            return {
+                ...prevState,
+                isCopied: true,
+            };
+        });
+
+        setTimeout(() => {
+            setState(prevState => {
+                return {
+                    ...prevState,
+                    isCopied: false,
+                };
+            });
+        }, 2000);
+    }
+
+    /**
+     * Handle inserting form into existing post/page
+     */
+    const handleInsert = async () => {
+
+        if (isButton && !state.openFormButton) {
+            openFormBtnRef.current?.focus();
+            return;
+        }
+
+        setState(prevState => {
+            return {
+                ...prevState,
+                isInserting: true,
+            };
+        });
+
+        const content = state?.posts[state.insertPostType]?.find((page) => page.value == state.selectedPost)?.content + getBlockComment();
+
+        await editEntityRecord('postType', state.insertPostType, state.selectedPost, {content});
+        const response = await saveEditedEntityRecord('postType', state.insertPostType, state.selectedPost, {content});
+
+        setState(prevState => {
+            return {
+                ...prevState,
+                isInserting: false,
+                isInserted: true,
+                insertedLink: response?.link,
+            };
+        });
+    };
+
+    /**
+     * Handle creating a new post/page
+     */
+    const handleCreateNew = async () => {
+        if (!state.newPostName) {
+            newPostNameRef.current?.focus();
+            return;
+        }
+
+        if (isButton && !state.openFormButton) {
+            openFormBtnRef.current?.focus();
+            return;
+        }
+
+        setState(prevState => {
+            return {
+                ...prevState,
+                isCreating: true,
+            };
+        });
+
+        const response = await saveEntityRecord('postType', state.createPostType, {
```

Comment:
Wow, it's pretty cool how much we can accomplish with just the the js wp api. 🪄

Reply:
I know. It's amazing :smile: